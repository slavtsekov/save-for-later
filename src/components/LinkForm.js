import React from 'react';

class LinkForm extends React.Component {
    constructor(props) {
        super();

        this.state = {
            title: props.link ? props.link.title : '',
            url: props.link ? props.link.url : '',
            description: props.link ? props.link.description : '',
            folder: props.link ? props.link.folder : '',
            error: undefined
        };
    }

    onFormSubmit = (e) => {
        e.preventDefault();
    };

    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({ title }));
    };

    onUrlChange = (e) => {
        const url = e.target.value;
        this.setState(() => ({ url }));
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };

    onFolderChange = (e) => {
        const folder = e.target.value;
        this.setState(() => ({ folder }));
    };

    render() {
        return (
            <form onSubmit={this.onFormSubmit}>
                {this.state.error && <p>{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Title"
                    value={this.state.title}
                    onChange={this.onTitleChange}
                />
                <input
                    type="text"
                    placeholder="URL"
                    value={this.state.url}
                    onChange={this.onUrlChange}
                />
                <textarea
                    placeholder="Add a description for the link (optional)"
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                >
                </textarea>
                <select
                    placeholder="Folder"
                    value={this.state.folder}
                    onChange={this.onFolderChange}
                >
                    <option value=""></option>
                </select>
            </form>
        );
    }
}

export { LinkForm as default };
