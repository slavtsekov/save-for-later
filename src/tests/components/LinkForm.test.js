import React from 'react';
import { shallow } from 'enzyme';
import LinkForm from '../../components/LinkForm';

test('should render LinkForm correctly', () => {
    const wrapper = shallow(<LinkForm />);
    expect(wrapper).toMatchSnapshot();
});

test('should set title on input change', () => {
    const wrapper = shallow(<LinkForm />);
    const value = 'Hot News';

    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });

    expect(wrapper.state('title')).toBe(value);
});

test('should set url on input change', () => {
    const wrapper = shallow(<LinkForm />);
    const value = 'http://test.te';

    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });

    expect(wrapper.state('url')).toBe(value);
});

test('should set description on input change', () => {
    const wrapper = shallow(<LinkForm />);
    const value = 'Check this later';

    wrapper.find('textarea').simulate('change', {
        target: { value }
    });

    expect(wrapper.state('description')).toBe(value);
});

test('should set folder on select change', () => {
    const wrapper = shallow(<LinkForm />);
    const value = 'Movies';

    wrapper.find('select').simulate('change', {
        target: { value }
    });

    expect(wrapper.state('folder')).toBe(value);
});
