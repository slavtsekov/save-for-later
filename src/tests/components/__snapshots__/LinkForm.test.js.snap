// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LinkForm correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Title"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="URL"
    type="text"
    value=""
  />
  <textarea
    onChange={[Function]}
    placeholder="Add a description for the link (optional)"
    value=""
  />
  <select
    onChange={[Function]}
    placeholder="Folder"
    value=""
  >
    <option
      value=""
    >
       --- 
    </option>
    <option
      key="folder1"
      value="folder1"
    >
      Books
    </option>
    <option
      key="folder2"
      value="folder2"
    >
      Games
    </option>
    <option
      key="folder3"
      value="folder3"
    >
      Movies
    </option>
  </select>
</form>
`;
